File: server/src/weblab/data/dto/Role.py
#!/usr/bin/python
# -*- coding: utf-8 -*-
#
# Copyright (C) 2005-2009 University of Deusto
# All rights reserved.
#
# This software is licensed as described in the file COPYING, which
# you should have received as part of this distribution.
#
# This software consists of contributions made by many individuals, 
# listed below:
#
# Author: Pablo Orduña <pablo@ordunya.com>
#

class Role(object):
    
    def __init__(self, name):
        super(Role, self).__init__()
        self.name = name
        
    def __repr__(self):
        return "Role(name = '%s')" % (
                self.name
            )

File: gwt-client/src/es/deusto/weblab/client/dto/users/Role.java
/*
* Copyright (C) 2005 onwards University of Deusto
* All rights reserved.
*
* This software is licensed as described in the file COPYING, which
* you should have received as part of this distribution.
*
* This software consists of contributions made by many individuals, 
* listed below:
*
* Author: Jaime Irurzun <jaime.irurzun@gmail.com>
*
*/

package es.deusto.weblab.client.dto.users;

public class Role {
	
	private String name;

	public Role() {}
	
	public Role(String name) {
		super();
		this.name = name;
	}

	public String getName() {
		return this.name;
	}
	public void setName(String name) {
		this.name = name;
	}
	
	@Override
	public boolean equals(Object other) {
		if ( other instanceof Role ) {
			return this.name.equals(((Role)other).name);
		} else {
			return false;
		}
		
	}
}

